### PR TITLE
[feat] add JumpStart foundation model text generation with few shot learning demo

### PR DESCRIPTION
**Pull request by Neo9061**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [ ] I have tested my notebook(s) and ensured it runs end-to-end
- [ ] I have linted my notebook(s) and code using `black-nb -l 100 {path}/{notebook-name}.ipynb`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: introduction_to_amazon_algorithms/jumpstart-foundation-models/text-generation-few-shot-learning.ipynb [hunk 1]
```diff
@@ -0,0 +1,1896 @@
+{
```
Comment (khetan2) [inline introduction_to_amazon_algorithms/jumpstart-foundation-models/text-generation-few-shot-learning.ipynb:1]: <p>There are plenty of other text generation models like <a href="https://huggingface.co/bigscience/bloom-7b1" target="_blank"><strong>Bloom 7b1</strong></a>, <a href="https://huggingface.co/bigscience/bloom-3b" target="_blank"><strong>Bloom 3b</strong></a>, <a href="https://huggingface.co/EleutherAI/gpt-neo-2.7B" target="_blank"><strong>GPT-NEO-2b7</strong></a>, and etc. --&gt; remove "plenty of"</p><p> </p><p>"make predictions thanks to only a couple of examples" needs rewriting.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3832/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (khetan2) [inline introduction_to_amazon_algorithms/jumpstart-foundation-models/text-generation-few-shot-learning.ipynb:1]: <p>Do we need session?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3832/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

File: introduction_to_amazon_algorithms/jumpstart-foundation-models/text2text-generation-flan-t5.ipynb [hunk 2]
```diff
@@ -934,6 +934,115 @@
     "    print(f\"{bold} The reasoning result is{unbold}: '{generated_texts}'{newline}\")"
```
Comment (khetan2) [inline introduction_to_amazon_algorithms/jumpstart-foundation-models/text2text-generation-flan-t5.ipynb:934]: <pre class="ql-syntax" spellcheck="false">Line #10.        "top_p": 0.95,
</pre><p>Please check how top_p and top_k work together. I believe one supersedes the other, and specifying both together does not help.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3832/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (khetan2) [inline introduction_to_amazon_algorithms/jumpstart-foundation-models/text2text-generation-flan-t5.ipynb:934]: <pre class="ql-syntax" spellcheck="false">Line #4.        "max_length": 5000,
</pre><p>Isn't this too large? Does the model even support 5000 tokens?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3832/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='56'/>

File: introduction_to_amazon_algorithms/jumpstart-foundation-models/text-generation-few-shot-learning.ipynb (hunk 1 above)
Comment (Neo9061) [reply introduction_to_amazon_algorithms/jumpstart-foundation-models/text-generation-few-shot-learning.ipynb:1]: <p>not needed. Have deleted. THX</p>

File: introduction_to_amazon_algorithms/jumpstart-foundation-models/text2text-generation-flan-t5.ipynb (hunk 2 above)
Comment (Neo9061) [reply introduction_to_amazon_algorithms/jumpstart-foundation-models/text2text-generation-flan-t5.ipynb:934]: <p>This has been answered by Kyle in the science chat room</p>

Comment (Neo9061) [reply introduction_to_amazon_algorithms/jumpstart-foundation-models/text2text-generation-flan-t5.ipynb:934]: <p>It does, i noticed BloomZ can handle such thing. I reduced to 1000.</p>

**Review by atqy — APPROVED** [review]

**Pull request merged**